### PR TITLE
Support runtime active effects

### PR DESCRIPTION
**Pull request by tguerin**
This pr aims to improve 3 things:
* It was possible to add an effect at runtime but not to remove it...
* The didUpdate was too brutal, now we can detect if effect is removed (also applies to post effect)
* Runtime effects are now ignored from the didUpdate widget. An effect added at runtime must be removed at runtime.

### CODE REVIEW

**Comment by tguerin** [comment]
@Joshix-1 wdyt? Does it fix your issue? Need to clean the code btw.

**Review by Joshix-1 — COMMENTED** [review]
I tested this, and it doesn't work

File: lib/src/newton.dart [hunk 1]
```diff
@@ -149,7 +151,9 @@ class NewtonState extends State<Newton> with SingleTickerProviderStateMixin {
   @override
   void didUpdateWidget(Newton oldWidget) {
     super.didUpdateWidget(oldWidget);
-    _setupEffectsFromWidget();
+    if (widget.activeEffects != oldWidget.activeEffects) {
```
[the remark this answers is not in the document]
Comment (tguerin) [reply lib/src/newton.dart:154]: cause you need to reuse the same instance if you want to keep track of already emitted particles. If you create another effect instance it's normal that the effect is replaced. I tested on the chrome navigator to resize the screen. When resized the effect keep animating (need to fix update surface size to have things work smoothly)

File: lib/src/newton.dart [hunk 2]
```diff
@@ -54,6 +54,8 @@ class NewtonState extends State<Newton> with SingleTickerProviderStateMixin {
   int _lastElapsedMillis = 0;
   final List<Effect> _activeEffects = List.empty(growable: true);
   final List<Effect> _pendingActiveEffects = List.empty(growable: true);
+  final List<Effect> _runtimeActiveEffects = List.empty(growable: true);
```
[the remark this answers is not in the document]
Comment (tguerin) [reply lib/src/newton.dart:57]: ah my bad I did that quickly yesterday night, should be merged with active effects in the painter. Will do that after work.

File: lib/src/newton.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (Joshix-1) [reply lib/src/newton.dart:154]: if you create the list in build the list is different everytime even with the same effects. That's probably unexpected.

**Comment by Joshix-1** [comment]
I feel like the solution of only allowing either dynamically added via NewtonState or fixed added via Newton arg is a better solution. Or maybe have two different widgets for each use case.
This feels like it's getting too complex

**Comment by tguerin** [comment]
As i said, i'm totally fine to remove the runtime version and stick to the effects passed as parameters. Will remove the runtime support. Thanks for your time!

**Comment by tguerin** [comment]
I reworked the PR and updated the description. I don't think we need 2 different widgets. A Newton widget can accept effects added at runtime howevert they must be removed at runtime (or automatically removed if there is a finite particle count). Still draft cause I need to test it properly.

**Comment by Joshix-1** [comment]
Code looks good

**Pull request merged**